### PR TITLE
Purs 0.15

### PR DESCRIPTION
**Pull request by iostat**
Needed for Chanterelle.

Mostly routine, deps + CommonJS FFI.
Added purs-tidy and tidy-check to CI, as well as bumping the CI base image.

PureScript's change to ESM actually broke dynamically loading solc. There's a hack in here that requires Node v20.6+ to work, but it works. It did require a slight API change to `L.S.C.useCompiler`

Added tests

### CODE REVIEW

**Comment by kejace** [comment]
👍

File: package.json [hunk 1]
```diff
@@ -14,8 +16,9 @@
     "solc": "^0.5"
```
Comment (martyall) [inline package.json]: what about this solc dep?

Comment (iostat) [reply package.json]: package.json deps don't matter for purescript -- it's worth noting that this library allows you to invoke /any/ version of the solidity compiler (see the FFI in Compiler.js) -- the API was just stabilized in 0.5+

Comment (iostat) [reply package.json]: (Releases.purs fetches/loads whatever version you want)

**Review by martyall — APPROVED** [review]

**Comment by martyall** [comment]
lgtm, excited about the tests

**Review by martyall — APPROVED** [review]

**Pull request merged**